### ISSUE TITLE
Automatically close Git Lens Terminal after command execution

### ISSUE THREAD

**Issue by muhammadyusuf-kurbonov** [gitkraken/vscode-gitlens#2312]
Now when I'm merging a branch, deleting it, or something else, Git Lens opens a new terminal panel, where it is integrated (GitLens Shell Integration). This opened terminal left after the command and sometimes I'm running other commands (such as `npm run serve`). 
And next time I'll use the GitLens command it can not be executed. 

So, it would be better if it will close automatically. How it is made in Docker extension.

P.s. great extension and features :+1: . Author :star:

**Comment by longrunningprocess** [comment]
agreed, I was just coming out here to find an option to turn that off but I didn't see one... I'd like this too.

**Comment by bradlis7** [comment]
I'd prefer it to just execute it in an invisible shell and put in in OUTPUT if that's what some users want. Why is using a terminal required?

**Comment by mscottx88** [comment]
Happy Leap Day! I was hoping not to find an open issue on this. I agree that it would be really nice to auto-close the auto-opened terminal because it is quite annoying to find yourself entering commands in the terminal you didn't want.

**Comment by 0reo** [comment]
Commenting out this line would stop the terminal from stealing focus, so you don't start accidentally typing into the wrong one

https://github.com/gitkraken/vscode-gitlens/blob/4cbc432bc07dc0d5340c83604ee7db3b5b90b2cf/src/env/node/git/git.ts#L2242

Can manually apply this change locally by going into `%USERPROFILE%\.vscode\extensions\eamodio.gitlens-15.0.3\dist\gitlense.js`(or wherever you've got the extension installed), searching for `d.show(!1)`, and just deleting it.

@eamodio any obvious issues with this change?  Or maybe this could be a setting if it's undesired to do this by default?

**Comment by aliaksandrsen** [comment]
for GitLess v11.7.2 I deleted `B.show(!1),` from `~/.vscode/extensions/maattdd.gitless-11.7.2/dist/gitlens.js`

**Comment by shinyypig** [comment]
> for GitLess v11.7.2 I deleted `B.show(!1),` from `~/.vscode/extensions/maattdd.gitless-11.7.2/dist/gitlens.js`

just search `.show(!1),`, in my case, it is `d.show(!1),`

**Comment by elkinjosetm** [comment]
+1

**Comment by SavenkovIgor** [comment]
Eeven 3 years later it is still a great thing to do

**Comment by bdemirel** [comment]
Yes please, this is super annoying

**Comment by eamodio** [comment]
Closing as we no longer use the terminal to execute Git commands
